### ISSUE TITLE
Getting SocketTimeoutException after creating connector

### ISSUE THREAD

**Issue by kedarr** [confluentinc/kafka-connect-elasticsearch#189]
I am facing issues with my kafka-connect-elasticsearch setup. I am trying to read data (Json data ) from kafka and trying to index it in elasticsearch. As soon as a I register a connector, I am getting SocketTimeoutException. Here is my connector configuration 

```
  batch.size = 200
  connection.url = [http://elasticsearch:9200]
  flush.timeout.ms = 10000
  key.ignore = true
  linger.ms = 1
  max.buffered.records = 1000
  max.in.flight.requests = 5
  max.retries = 5
  retry.backoff.ms = 100
  schema.ignore = true
  topic.index.map = [topic1:estopic1]
  topic.key.ignore = []
  topic.schema.ignore = []
  type.name = topic1-type
```

Docker image versions : 

- Kafka - confluentinc/cp-kafka:latest
- Zookeeper - image: confluentinc/cp-zookeeper:latest
- Elasticsearch - docker.elastic.co/elasticsearch/elasticsearch:6.2.3
- Connect - confluentinc/cp-kafka-connect:latest . Also tried with confluentinc/cp-kafka-connect:3.3.0

Stacktrace 

```
[2018-03-28 01:20:30,746] ERROR Task kafka-es-connector-1-0 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask)
org.apache.kafka.connect.errors.ConnectException: java.net.SocketTimeoutException: Read timed out
	at io.confluent.connect.elasticsearch.ElasticsearchWriter.createIndicesForTopics(ElasticsearchWriter.java:254)
	at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.open(ElasticsearchSinkTask.java:113)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.openPartitions(WorkerSinkTask.java:482)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.access$1000(WorkerSinkTask.java:55)
	at org.apache.kafka.connect.runtime.WorkerSinkTask$HandleRebalance.onPartitionsAssigned(WorkerSinkTask.java:518)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.onJoinComplete(ConsumerCoordinator.java:265)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:363)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:310)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:297)
	at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1078)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1043)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.pollConsumer(WorkerSinkTask.java:366)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:246)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:180)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:148)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:146)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:190)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:170)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:261)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:165)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:272)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:124)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:271)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
	at io.searchbox.client.http.JestHttpClient.execute(JestHttpClient.java:48)
	at io.confluent.connect.elasticsearch.ElasticsearchWriter.createIndicesForTopics(ElasticsearchWriter.java:249)
	... 21 more
[2018-03-28 01:20:30,746] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask)
[2018-03-28 01:20:30,746] INFO Stopping ElasticsearchSinkTask. (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
```

**Comment by yuanman** [comment]
I have same problem too.

**Comment by dgaikwad** [comment]
I am also facing the same problem.

**Comment by maarab7** [comment]
Did anyone find solution for this?

**Comment by dgaikwad** [comment]
@maarab7  try with deploying multiple kafka-connector.

**Comment by maarab7** [comment]
> @maarab7 try with deploying multiple kafka-connector.

Did this work for you?

**Comment by dgaikwad** [comment]
@maarab7 yes

**Comment by zhouxianggen** [comment]
this may be caused by connection timeout (connector to es), try set config of your connector:
"connection.timeout.ms": 5000,
"read.timeout.ms": 5000,

then restart you connector.

**Comment by kujon** [comment]
This is still super flaky. And when it hits it necessitates restarting the tasks manually.

**Comment by levzem** [comment]
See #381 for discussion on possible solutions. Closing now.

**Comment by avgojzack** [comment]
Has this been included in a release yet? I see the PR merged in February, but I don't see the fix in 5.4.x or 5.5.x.

**Comment by levzem** [comment]
@avgojzack this was merged to master and unfortunately it didn't make it before the `5.5.x` branch was cut, so you can expect this to make into `6.0.x`. If you need the fix now, you can download and use the snapshot `master` version.
